nservicebus/azure-service-bus/configuration/configuration.md

```diff
@@ -0,0 +1,200 @@
+
+### Messaging Factories
+
+Messaging factories are the heart of connectivity management in the Azure Service Bus SDK, each messaging factory maintains a TCP connection with the broker and creates MessageSender and MessageReceiver instances for that connection. This implies that all senders and receivers created by the same factory, use the same underlying TCP connection and inherit the settings configured at the messaging factory level. The following settings allow to control the messaging factories.
+
+ * `NumberOfMessagingFactoriesPerNamespace(int)`: NServiceBus maintains a small pool of messaging factories per namespace, this setting determines the size of the pool. Defaults to 5.
+ * `RetryPolicy(RetryPolicy)`: Determines how entities should respond on transient connectivity failures. Defaults to `RetryPolicy.Default`, which is an exponential retry under the hood.
+ * `BatchFlushInterval(TimeSpan)`: This setting controls the batching behavior for message senders. They will buffer send operations during this time frame and send all messages at once. Defaults to 0.5 seconds. Specify TimeSpan.Zero to turn batching off.
+ * `MessagingFactorySettingsFactory(Func<string, MessagingFactorySettings>)`: This factory method allows to override creation of messaging factories.
+ 
+### Message Receivers
+
+The following settings determine how NServiceBus will receive messages from Azure Service Bus entities.
+
+ * `ReceiveMode(ReceiveMode)`: Determines part of the 'transactional behavior'. `ReceiveMode.PeekLock` ensure that messages will be only removed from the queue after processing is done. If processing fails, or takes to long, the message will reappear on the queue for re-processing. Basically emulating transactional rollback. `ReceiveMode.ReceiveAndDelete` will delete the message immediately after receive, meaning there is no rollback when an exception occurs. Defaults to `ReceiveMode.PeekLock`.
+ * `PrefetchCount(int)`: This setting will make the receiver prefetch messages on receive operations, and acts like client side batching. Defaults to 200, meaning that each receive operation will pull in another 200 messages.
+ * `RetryPolicy(RetryPolicy)`: Determines how the receiver should respond on transient connectivity failures. Defaults to `RetryPolicy.Default`, which is an exponential retry under the hood.
+ * `AutoRenewTimeout(TimeSpan)`: When using `ReceiveMode.PeekLock`, the broker will lock a received message for a period specified as `LockDuration(TimeSpan)` on the entity. If processing doesn't end within this time period, a message will reappear on the entity and will be reprocessed. For long running operations this might imply that the operation will be executed multiple times in parallel, which is usually undesirable. This setting allows to extend the lock automatically when processing hasn't finished yet. Ensure that the `TimeSpan` specified here is a little smaller than the `TimeSpan` specified as `LockDuration`.
```

Comment:
In my opinion: we should create an article about how to do long running operations instead of just issuing a warning here. This should not be abused to do long running operations properly.